types/pg/index.d.ts

```diff
@@ -208,12 +217,12 @@ export interface PoolClient extends ClientBase {
     release(err?: Error): void;
 }
 
-export class Query extends events.EventEmitter implements Submittable {
-    constructor(queryTextOrConfig?: string | QueryConfig, values?: any[]);
+export class Query<R, I extends any[]> extends events.EventEmitter implements Submittable {
```

Comment:
`ResultBuilder` requires `R` to extend `QueryResultRow`. Therefore, since `Query` receives an `R` and passes it to `ResultBuilder`, it should also enforce it to extend `QueryResultRow`. I believe the current code leads to a compilation error, but even if it doesn't, I would suggest to establish the aforementioned type restriction.

Reply:
Strangely enough it doesn't generate a compile error, maybe that's due to the `any` default type? But you're right that constraint should be added (and I have now).

I also completely forgot to add defaults here...